src/Ersatz/Orderable.hs

```diff
@@ -57,6 +66,44 @@ instance Orderable Bit where
   a <?  b = not a && b
   a <=? b = not a || b
 
+-- | Compare by lexicographic order on sorted key-value pairs
+instance (Ord k, Orderable v) => Orderable (Map.Map k v) where
+  x <?  y = assocsLt (Map.assocs x) (Map.assocs y)
+  x <=? y = assocsLe (Map.assocs x) (Map.assocs y)
+
+-- | Compare by lexicographic order on sorted key-value pairs
+instance Orderable v => Orderable (IntMap.IntMap v) where
+  x <?  y = assocsLt (IntMap.assocs x) (IntMap.assocs y)
+  x <=? y = assocsLe (IntMap.assocs x) (IntMap.assocs y)
+
+assocsLt :: (Ord k, Orderable v) => [(k,v)] -> [(k,v)] -> Bit
+assocsLt _ [] = false
+assocsLt [] _ = true
+assocsLt ((k1,v1):xs) ((k2,v2):ys) =
+  case compare k1 k2 of
+    LT -> true
+    GT -> false
+    EQ -> v1 <? v2 || v1 === v2 && assocsLt xs ys
+
+assocsLe :: (Ord k, Orderable v) => [(k,v)] -> [(k,v)] -> Bit
+assocsLe [] _ = true
+assocsLe _ [] = false
+assocsLe ((k1,v1):xs) ((k2,v2):ys) =
+  case compare k1 k2 of
+    LT -> true
+    GT -> false
+    EQ -> v1 <? v2 || v1 === v2 && assocsLe xs ys
```

Comment:
Should this be `<=?` instead of `<?`?

Reply:
I don't think so, the only difference in `<` and `<=` on lists is what happens when you get to the end of the list. You'll noticed that `instance Orderable []` has this same structure. Only the *recursive* call uses `<=?` (corresponding to the recursive call to `assocsLe`

Reply:
Incidentally, I considered making a `newtype Assoc k v = Assoc (k,v)` and then using `Data.Coerce.coerce :: [(k,v)] -> [Assoc k v]` and then using an `instance (Ord k, Orderable v) => Orderable (Assoc k v)` together with `instance Orderable a => Orderable [a]` but that felt a bit too heavy for what I was actually doing here.

Reply:
Fair enough, being consistent with the `Orderable` instance for lists is a compelling argument.